cmake/kokkos_enable_options.cmake

```diff
@@ -23,6 +23,11 @@ ENDFUNCTION()
 KOKKOS_CFG_DEPENDS(OPTIONS DEVICES)
 KOKKOS_CFG_DEPENDS(OPTIONS COMPILER_ID)
 
+# set the default build type if not provided
+if("${CMAKE_BUILD_TYPE}" STREQUAL "")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "CMake build type" FORCE)
```

Comment:
I'm not sure about this. There is good reason to allow an empty build type. People could pass custom optimization/debug flags to `CMAKE_CXX_FLAGS` and these might get overwritten by the release flags.

Reply:
I would also make this part of another PR, not the Python stuff here.

Reply:
Fair enough about the PR. But they could always just set it to one of the existing modes and then change `CMAKE_CXX_FLAGS_<BUILD_TYPE>`. I personally have made the mistake of compiling and running Kokkos without setting the build type so I think it is safe to assume that there might be a fair number of users who don't realize their code is running _much_ slower than it should be.